### PR TITLE
feat(webpack): auto detect core-js version

### PR DESCRIPTION
**Pull request by farnabaz**
Provide easy way to change `core-js` version and auto detect `core-js` installed version

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/webpack/src/config/base.js [hunk 1]
```diff
@@ -96,6 +97,8 @@ export default class WebpackBaseConfig {
 
   getBabelOptions () {
     const envName = this.name
+    const { corejs = 'auto' } = this.buildContext.buildOptions
```
Comment (pi0) [inline packages/webpack/src/config/base.js:100]: We should add defaults to `@nuxt/config/src/config/build.json` (also update types with `yarn jest packages/config -u`)

File: packages/webpack/src/config/base.js [hunk 2]
```diff
@@ -96,6 +97,8 @@ export default class WebpackBaseConfig {
...
+    const { rootDir } = this.buildContext.options
```
Comment (pi0) [inline packages/webpack/src/config/base.js:101]: Refactor: we may destructure both in single line :)

**Review by clarkdo — DISMISSED** [review]

File: packages/webpack/src/config/base.js [hunk 3]
```diff
@@ -113,8 +115,11 @@ export default class WebpackBaseConfig {
         }
       )
     }
-
-    const defaultPreset = [require.resolve('@nuxt/babel-preset-app'), {}]
+    const defaultPreset = [require.resolve('@nuxt/babel-preset-app'), {
+      corejs: {
+        version: corejs === 'auto' ? createRequire(rootDir)('core-js').version[0] : corejs
```
Comment (pi0) [inline packages/webpack/src/config/base.js:120]: What do you think to compute `version` outside (L118) and also do a validation to ensure it is either `"2"` or `"3"`?

Comment (farnabaz) [reply packages/webpack/src/config/base.js]: What we do on invalid values? choose default value or raise an error?

Comment (pi0) [reply packages/webpack/src/config/base.js]: A warn could be enough :)

https://babeljs.io/docs/en/babel-preset-env#corejs

Added here: https://github.com/nuxt/nuxt.js/pull/7785/commits/4424e9c61556b5005f6955d619ddc99c00349ba5#diff-d8071aa99dfe30c3c1c6135dfd903334R129

Comment (farnabaz) [reply packages/webpack/src/config/base.js]: 👍 ❤️

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
@farnabaz Would be nice also adding to the docs

**Comment by rchl** [comment]
a) I don't see types being updated
b) It seems like the code is assigning strings rather than numbers to `corejs.version`. Won't this be a problem for babel-preset-env?

File: packages/webpack/src/config/base.js [hunk 4]
```diff
@@ -114,7 +116,26 @@ export default class WebpackBaseConfig {
...
+    let corejsVersion = corejs
+    if (corejsVersion === 'auto') {
+      try {
+        corejsVersion = createRequire(rootDir)('core-js').version.split('.')[0]
+      } catch (_err) {
+        corejsVersion = '2'
+      }
+    }
+
+    if (corejsVersion !== '2' && corejsVersion !== '3') {
```
Comment (rchl) [inline packages/webpack/src/config/base.js:129]: Comparing with strings but user is supposed to provide 2 or 3 as numbers?

**Comment by pi0** [comment]
@rchl It is normalized and passed as a string [src](https://github.com/babel/babel/blob/master/packages/babel-preset-env/src/normalize-options.js#L242) 

Would be nice if can open a PR to support number and also add missing types :)

**Pull request merged**